spec.bs

```diff
@@ -917,6 +966,28 @@ To <dfn>validate and convert auction ad config</dfn> given an {{AuctionAdConfig}
       1. Set |auctionConfig|'s [=auction config/seller signals=] to the result of
         [=serializing a JavaScript value to a JSON string=], given |result|.
     * To handle an error, set |auctionConfig|'s [=auction config/seller signals=] to failure.
+1. If |config|["{{AuctionAdConfig/auctionNonce}}"] [=map/exists=]:
+  1. Let |nonce| be an empty [=string=].
+  1. [=list/For each=] |i| in [=the range=] from 0 to the [=string/length=] of
+    |config|["{{AuctionAdConfig/auctionNonce}}"], exclusive:
+    1. Let |c| be the |i|th [=code unit=] of |config|["{{AuctionAdConfig/auctionNonce}}"] if it is
+      0x0000 NULL or in the range 0x0020 SPACE to 0x007F DELETE, inclusive, otherwise "`?`".
+    1. Append |c| to the end of |nonce|.
+  1. Set |auctionConfig|'s [=auction config/auction nonce=] to the result of running
+    [=get uuid from string=] with |nonce|.
+1. Let |additionalBids| be null.
+1. If |config|["{{AuctionAdConfig/additionalBids}}"] [=map/exists=]:
+  1. If |config|["{{AuctionAdConfig/componentAuctions}}"]'s [=list/size=] is greater than 0, or
+    |config|["{{AuctionAdConfig/auctionNonce}}"] does not [=map/exist=], [=exception/throw=] a
+    {{TypeError}}.
+  1. Set |auctionConfig|'s [=auction config/expects additional bids=] to true.
+  1. Set |additionalBids| to a new [=list=] of [=generated bids=].
+  1. [=Handle an input promise in configuration=] given |auctionConfig| and
+    |config|["{{AuctionAdConfig/additionalBids}}"]:
+    * To parse the value |result|:
+      1. Nothing to do.
+    * To handle an error: (TODO: check if this is correct when the promise is rejected.)

```

Comment:
I went looking for how we've built this. As far as I can tell, a Promise successfully resolved triggers the rest of the behavior below; a Promise rejected is quietly dropped. (https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/modules/ad_auction/navigator_auction.cc;drc=49a952fddb576b129aab7c6b333513b0d9e22adf;l=1502)

Reply:
Promise rejected aborts the auction:
https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/modules/ad_auction/navigator_auction.cc;drc=49a952fddb576b129aab7c6b333513b0d9e22adf;l=2787

[=Handle an input promise in configuration=] ought to be handling that, but I am not sure we have a mechanism for that.

You should probably set the field to failure and check for that? I should probably double-check that all the places that look at the promises count are correct, too...

Reply:
I think it should be handled here in [Handle an input promise in configuration](https://github.com/WICG/turtledove/blob/main/spec.bs#L3368). Current it only says "Execute the steps for error-handling" which does not seem very clear what steps will be executed. Better to improve there by saying fail the auction, and maybe we can put the steps that happens when an auction fails to a separate function and then we can link to that method there, in a separate PR.

And I think what to set for the field here does not matter that much.

Reply:
I remembered how it works. Basically, the other promise's "handle error sections" set the fields to false, then this check:
https://pr-preview.s3.amazonaws.com/qingxinwu/turtledove/pull/796.html#wait-until-scoring-dependency-done:~:text=If%20auctionConfig%E2%80%99s%20auction%20signals%2C%20seller%20signals%2C%20per%20buyer%20signals%2C%20per%20buyer%20currencies%20or%20per%20buyer%20timeouts%20is%20failure%2C%20return%20failure.

Returns failure, and then the caller of that fails things.

Reply:
oh right, I reworked this part a little bit using some little hack, adding a new state `null` to `expects additional bids`. PTAL.